dask/dataframe/core.py

```diff
@@ -2415,6 +2415,11 @@ def to_series(self):
         return self.map_partitions(M.to_series,
                                    meta=self._meta.to_series())
 
+    @derived_from(pd.Index)
+    def to_frame(self, name=None):
+        return self.map_partitions(M.to_frame, True, name,
+                                   meta=self._meta.to_frame(True, name))
```

Comment:
we'll likely need a compatibility layer here for older versions of pandas where `name` is not allowed.

Reply:
Can you point me to an example please? I can do the rest

Reply:
I think we have a `PANDAS_VERSION` variable defined somewhere (`dataframe/utils.py` maybe?)

So something like

```
kwargs = {}

if PANDAS_VERSION <= '0.24.0' and name is not None:
    raise ValueError('not supported for ...')
elif PANDAS_VERSION >= '0.24.0':
    kwargs['name'] = name
 meta = self._meta.to_frame(True, **kwargs)
```